src/opensynth/models/faraday/gaussian_mixture/metrics.py

```diff
@@ -143,56 +132,22 @@ def update(
-            xm_prob = means * torch.matmul(responsibilities.t(), data)
-            covars = x_prob - 2 * xm_prob + m_prob
-            if self.covariance_type == "diag":
-                self.covariance_sum.add_(covars)
-            else:  # covariance_type == "spherical"
-                self.covariance_sum.add_(covars.mean(1))
-        elif self.covariance_type == "tied":
-            # This is taken from
-            # https://github.com/scikit-learn/scikit-learn/blob/
-            # 844b4be24d20fc42cc13b957374c718956a0db39/sklearn/mixture/
-            # _gaussian_mixture.py#L183
-            x_sq = data.T.matmul(data)
-            mean_sq = (data_component_weights * means.T).matmul(means)
-            self.covariance_sum.add_(x_sq - mean_sq)
-        else:  # covariance_type == "full":
-            # We iterate over each component since this is typically faster...
-            for i in range(self.num_components):
-                component_diff = data - means[i]
-                covars = (
-                    responsibilities[:, i].unsqueeze(1) * component_diff
-                ).T.matmul(component_diff)
-
-                # Add regularization to the diagonal of the covariance matrix
-                regularization = self.reg * torch.eye(
-                    self.num_features, device=covars.device
-                )
-                covars_regularized = covars + regularization
-
-                self.covariance_sum[i].add_(covars_regularized)
+        # We iterate over each component since this is typically faster...
+        for i in range(self.num_components):
+            component_diff = data - means[i]
+            covars = (
+                responsibilities[:, i].unsqueeze(1) * component_diff
+            ).T.matmul(component_diff)
 
-    def compute(self) -> torch.Tensor:
-        if self.covariance_type == "diag":
-            return (
-                self.covariance_sum / self.component_weights.unsqueeze(-1)
-                + self.reg
+            # Add regularization to the diagonal of the covariance matrix
+            regularization = self.reg * torch.eye(
+                self.num_features, device=covars.device
             )
-        if self.covariance_type == "spherical":
-            return (
-                self.covariance_sum / self.component_weights
-                + self.reg * self.num_features
-            )
-        if self.covariance_type == "tied":
-            result = self.covariance_sum / self.component_weights.sum()
-            shape = result.size()
-            result = result.flatten()
-            result[:: self.num_features + 1].add_(self.reg)
-            return result.view(shape)
+            covars_regularized = covars + regularization
+
+            self.covariance_sum[i].add_(covars_regularized)
+
+    def compute(self) -> torch.Tensor:
         # covariance_type == "full"
         result = self.covariance_sum / self.component_weights.unsqueeze(
             -1
```

Comment:
Remove k-means metrics (line 168 onwards) as they are no longer needed for sk-learn implementation.